sentry-android-core/src/main/java/io/sentry/android/core/SentryAndroid.java

```diff
@@ -102,11 +102,24 @@ public static synchronized void init(
                 (isTimberUpstreamAvailable
                     && classLoader.isClassAvailable(SENTRY_TIMBER_INTEGRATION_CLASS_NAME, options));
 
-            AndroidOptionsInitializer.init(
-                options, context, logger, isFragmentAvailable, isTimberAvailable);
+            final BuildInfoProvider buildInfoProvider = new BuildInfoProvider(logger);
+            final LoadClass loadClass = new LoadClass();
+
+            AndroidOptionsInitializer.loadDefaultAndMetadataOptions(
+                options, context, logger, buildInfoProvider);
+
             configuration.configure(options);
-            deduplicateIntegrations(options, isFragmentAvailable, isTimberAvailable);
             resetEnvelopeCacheIfNeeded(options);
```

Comment:
I guess we don't need this one anymore, if we set the AndroidEnvelopeCache after the user has configured the options

Reply:
In terms of integrations we still do, as both Fragment and Timber integration could be duplicates, and since there are tests for this I know we could still have this situation.
But I agree, since the user can add the integrations first we could only add the Sentry-default ones in case they don't exist, making this method obsolete

Reply:
no, I meant `resetEnvelopeCacheIfNeeded`, the deduplicate method we still need, yes :)

Reply:
Oh right, good point 👍